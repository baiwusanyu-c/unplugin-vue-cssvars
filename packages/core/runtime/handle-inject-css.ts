import { parserCompiledSfc } from '../parser'
import { injectCSSVars } from '../inject'
import type { MagicStringBase } from 'magic-string-ast'
import type { IVueCSSVarsCtx } from '../types'

export function handleInjectCss(
  id: string,
  code: string,
  mgcStr: MagicStringBase,
  ctx: IVueCSSVarsCtx,
) {
  const parseRes = parserCompiledSfc(code)
  debugger
  const injectRes = injectCSSVars(
    ctx.vbindVariableList.get(id),
    ctx.isScriptSetup,
    parseRes,
    mgcStr,
  )
  mgcStr = injectRes.mgcStr
  injectRes.vbindVariableList && ctx.vbindVariableList.set(id, injectRes.vbindVariableList)
  return mgcStr
}
