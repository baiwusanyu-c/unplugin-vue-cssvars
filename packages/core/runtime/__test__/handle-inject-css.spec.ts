import { describe, expect, test } from 'vitest'
import MagicString from 'magic-string'
import { handleInjectCss } from '../handle-inject-css'
import code from './sfc/mock-code.txt?raw'
describe('handle inject css', () => {
  test('handleInjectCss: basic', () => {
    const mockId = 'D:/project-github/unplugin-vue-cssvars/play/vite/src/views/app/App.vue'
    const mockCode = code
    const mockMgc = new MagicString(code)
    const mockContext = {
      vbindVariableList: new Map(),
      isScriptSetup: true,
    }
    mockContext.vbindVariableList.set(mockId, [
      { value: 'sassColor', has: true, isRef: true },
      { value: 'color', has: true, isRef: true },
    ])
    const res = handleInjectCss(mockId, mockCode, mockMgc, mockContext as any)
    expect(res).toMatchSnapshot()
  })
})
