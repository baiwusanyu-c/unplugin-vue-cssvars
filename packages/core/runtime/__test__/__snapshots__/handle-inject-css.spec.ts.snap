// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`handle inject css > handleInjectCss: basic 1`] = `MagicString {}`;
